src/main/java/com/amazonaws/services/sqs/AmazonSQSIdleQueueDeletingClient.java

```diff
@@ -189,17 +189,26 @@ public CreateQueueResult createQueue(CreateQueueRequest request) {
         if (!queueName.startsWith(queueNamePrefix)) {
             throw new IllegalArgumentException();
         }
-
+        String retentionPeriodString = retentionPeriod.get().toString();
+        int retry = 0;
         CreateQueueRequest superRequest = request.clone()
                 .withQueueName(queueName)
-                .withAttributes(attributes);
+                .withAttributes(attributes)
+                .addTagsEntry(IDLE_QUEUE_RETENTION_PERIOD_TAG, retentionPeriodString);
 
         CreateQueueResult result = super.createQueue(superRequest);
         String queueUrl = result.getQueueUrl();
 
-        String retentionPeriodString = retentionPeriod.get().toString();
-        amazonSqsToBeExtended.tagQueue(queueUrl,
-                Collections.singletonMap(IDLE_QUEUE_RETENTION_PERIOD_TAG, retentionPeriodString));
+        // In case of throttling issues retry tagging the queue
+        while (retry++ < 5 && !listQueueTags(queueUrl).getTags().containsKey(IDLE_QUEUE_RETENTION_PERIOD_TAG)) {
```

Comment:
Retry strategies are a very rich and contentious source of configuration. I would strongly suggest you make use of the `com.amazonaws.retry` package in `aws-java-sdk-core` :) 

I'm also not convinced we want retrying at this level at all though - it adds complexity and another configuration knob (the retry policy above). Customers can do their own retry at the `CreateQueue` level on this client.

Reply:
Will take a look at `com.amazonaws.retry`, in the meantime removed retry for this section